stingray/lightcurve.py

```diff
@@ -664,38 +770,32 @@ def rebin(self, dt_new=None, f=None, method='sum'):
             raise ValueError("New time resolution must be larger than "
                              "old time resolution!")
 
-        if self.gti is None:
-
-            bin_time, bin_counts, bin_err, _ = \
-                utils.rebin_data(self.time, self.counts, dt_new,
-                                 yerr=self.counts_err, method=method)
-
-        else:
-            bin_time, bin_counts, bin_err = [], [], []
-            gti_new = []
-            for g in self.gti:
-                if g[1] - g[0] < dt_new:
-                    continue
-                else:
-                    # find start and end of GTI segment in data
-                    start_ind = self.time.searchsorted(g[0])
-                    end_ind = self.time.searchsorted(g[1])
+        bin_time, bin_counts, bin_err = [], [], []
+        gti_new = []
+        for g in self.gti:
+            if g[1] - g[0] < dt_new:
+                continue
```

Comment:
We might need some kind of special behaviour here if this condition is true, but also there's only a single GTI for some reason? This is probably rare, but it would be useful to know what the behaviour there is, and whether we want to throw a custom error to inform the user that that's what's happening?

Also, really, this should be `g[1] - g[0] < 3*dt_new` or something, no? Since the code will fail if there's only a single data point in a light curve?

Reply:
For the first, I added a ValueError if `gti_new` is empty at the end of the loop.
For the second, I think there might well be one bin in a given GTI, provided that the total light curve has more than one data. If you don't need power spectral analysis but just plotting, that single point might be useful.

Reply:
But there's a check earlier in the light curve creation that says:

```
if len(time) <= 1:
    raise StingrayError("A single or no data points can not create "
                                    "a lightcurve!")
```

So by default this will fail?

Reply:
Only if there is only one GTI. We might have multiple GTIs with one point per GTI and have a perfectly valid light curve with multiple time bins.